mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala

```diff
@@ -80,7 +80,7 @@ private[shared] object SharedParamsCodeGen {
       ParamDesc[String]("weightCol", "weight column name. If this is not set or empty, we treat " +
         "all instance weights as 1.0"),
       ParamDesc[String]("solver", "the solver algorithm for optimization. If this is not set or " +
-        "empty, default value is 'auto'", Some("\"auto\"")),
+        "empty, default value is 'auto'", finalFields = false),
```

Comment:
Why remove default value? Can we still keep a default value for the most common case?